File: test/fixtures/node-axios/multipart_with_headers.js
const axios = require('axios');
const FormData = require('form-data');
const fs = require('fs');

const form = new FormData();
form.append('attributes', '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}');
form.append('file', fs.readFileSync('myfile.jpg'), 'myfile.jpg');

const response = await axios.post(
    'http://localhost:28139/api/2.0/files/content',
    form,
    {
        headers: {
            ...form.getHeaders(),
            'Authorization': 'Bearer ACCESS_TOKEN',
            'X-Nice': 'Header'
        }
    }
);

File: test/fixtures/csharp/multipart_with_headers.cs
using System.Net.Http;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "http://localhost:28139/api/2.0/files/content");

request.Headers.Add("Authorization", "Bearer ACCESS_TOKEN");
request.Headers.Add("X-Nice", "Header");


MultipartFormDataContent content = new MultipartFormDataContent();
content.Add(new StringContent("{\"name\":\"tigers.jpeg\", \"parent\":{\"id\":\"11446498\"}}"), "attributes");
content.Add(new ByteArrayContent(File.ReadAllBytes("myfile.jpg")), "file", Path.GetFileName("myfile.jpg"));
request.Content = content;

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();
